### PR TITLE
[Bugfix] Fix error sql generated for jdbc oracle table

### LINKED ISSUE #9513: querying jdbc oracle external table with limit returns an error
### Steps to reproduce the behavior (Required)
1. CREATE RESOURCE
```sql
create external resource jdbc_test
properties (
"type"="jdbc",
"user"="xxx",
"password"="xxx",
"jdbc_uri"="xxx",
"driver_url"="https://repo1.maven.org/maven2/com/oracle/database/jdbc/ojdbc6/11.2.0.4/ojdbc6-11.2.0.4.jar",
"driver_class"="oracle.jdbc.driver.OracleDriver"
);
```
2. CREATE TABLE
```sql
create external table jdbc_tbl (`id` decimal(38,0) null, `data` varchar(200) null)
engine=jdbc properties (
"resource"="jdbc_test",
"table"="test"
);
```
3. QUERY
```sql
select * from jdbc_tbl limit 1;
```

### Expected behavior (Required)
return query result successfully
### Real behavior (Required)
```sql
ERROR 1064 (HY000): open JDBCScanner failed, error: java.sql.SQLSyntaxErrorException: ORA-00933: SQL 命令未正确结束
[oracle.jdbc.driver.T4CTTIoer.processError(T4CTTIoer.java:447), oracle.jdbc.driver.T4CTTIoer.processError(T4CTTIoer.java:396), oracle.jdbc.driver.T4C8Oall.processError(T4C8Oall.java:951), oracle.jdbc.driver.T4CTTIfun.receive(T4CTTIfun.java:513), oracle.jdbc.driver.T4CTTIfun.doRPC(T4CTTIfun.java:227), oracle.jdbc.driver.T4C8Oall.doOALL(T4C8Oall.java:531), oracle.jdbc.driver.T4CStatement.doOall8(T4CStatemen
```
### StarRocks version (Required)
2.3

### PR DESCRIPTION
**Pull request by silverbullet233**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #9513

### CODE REVIEW

File: be/src/connector/jdbc_connector.cpp [hunk 1]
```diff
@@ -49,7 +49,13 @@ static std::string get_jdbc_sql(const Slice jdbc_url, const std::string& table,
         }
     }
     if (limit != -1) {
-        oss << " LIMIT " << limit;
+        if (jdbc_url.starts_with("jdbc:oracle")) {
+            // oracle doesn't support limit clause, we should generate a subquery to do this
+            // ref: https://stackoverflow.com/questions/470542/how-do-i-limit-the-number-of-rows-returned-by-an-oracle-query-after-ordering
+            return fmt::format("SELECT * FROM ({}) WHERE ROWNUM <= {}", oss.str(), limit);
```
Comment (stdpain) [inline be/src/connector/jdbc_connector.cpp:55]: Are you sure we don't have to add an alias for subquery?

Comment (silverbullet233) [reply be/src/connector/jdbc_connector.cpp:55]: > Are you sure we don't have to add an alias for subquery?

I have tested it and it seems to be OK

**Review by kangkaisen — APPROVED** [review]

**Comment by kangkaisen** [comment]
Please add test cases

File: be/src/connector/jdbc_connector.cpp [hunk 2]
```diff
@@ -49,7 +49,13 @@ static std::string get_jdbc_sql(const Slice jdbc_url, const std::string& table,
         }
     }
     if (limit != -1) {
-        oss << " LIMIT " << limit;
+        if (jdbc_url.starts_with("jdbc:oracle")) {
```
Comment (imay) [inline be/src/connector/jdbc_connector.cpp:52]: case sensitive or case insensitive?

Comment (silverbullet233) [reply be/src/connector/jdbc_connector.cpp:52]: > case sensitive or case insensitive?

case sensitive

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Review by stdpain — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by silverbullet233** [comment]
https://github.com/Mergifyio backport branch-2.3

**Pull request merged**